public/app/features/alerting/unified/components/DynamicTable.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { FC, ReactNode } from 'react';
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme } from '@grafana/data';
+import { IconButton, useStyles, useTheme } from '@grafana/ui';
+import { useMedia } from 'react-use';
+
+export interface DynamicTableColumnProps<T = any> {
+  id: string | number;
+  label: string;
+
+  renderRow?: (item: DynamicTableItemProps<T>) => ReactNode;
+  size?: number | string;
+}
+
+export interface DynamicTableItemProps<T = any> {
```

Comment:
here comes the type police :oncoming_police_car:    
could we make it `T = unknown` for max safety?

Reply:
Yup yup